ext/storage_sources/gcp_store/test/test_gcp_connection.cpp

```diff
@@ -30,5 +30,338 @@
+    const bool list_single = true;
+    std::vector<std::string> objects;
+
+    google::cloud::storage::Client client = conn.get_client();
+
+    std::ofstream File(file_name);
+    std::string payload = "Test payload :)";
+    File << payload;
+    File.close();
+
+    SECTION("Simple list test", "[gcp-connection]")
+    {
+        const int32_t total_objects = 20;
+
+        // No matching objects. Object size should be 0.
+        REQUIRE(conn.list_objects(objects, false) == 0);
+        REQUIRE(objects.size() == 0);
+
+        // No matching objects with list_single. Object size should be 0.
+        REQUIRE(conn.list_objects(objects, list_single) == 0);
+        REQUIRE(objects.size() == 0);
+
+        // Upload 1 file and test list.
+        objects.clear();
+        client.UploadFile(
+          file_path, test_defaults::bucket_name, test_defaults::obj_prefix + object_key);
+        REQUIRE(conn.list_objects(objects, false) == 0);
+        REQUIRE(objects.size() == 1);
+
+        // Delete that object and test list.
+        objects.clear();
+        client.DeleteObject(test_defaults::bucket_name, test_defaults::obj_prefix + object_key);
+        REQUIRE(conn.list_objects(objects, false) == 0);
+        REQUIRE(objects.size() == 0);
+
+        // Upload multiple files and test list.
+        objects.clear();
+        for (int i = 0; i < total_objects; i++) {
+            client.UploadFile(file_path, test_defaults::bucket_name,
+              test_defaults::obj_prefix + std::to_string(i) + ".txt");
+        }
+        REQUIRE(conn.list_objects(objects, false) == 0);
+        REQUIRE(objects.size() == total_objects);
+
+        // Test if list single only returns one object.
+        objects.clear();
+        REQUIRE(conn.list_objects(objects, list_single) == 0);
+        REQUIRE(objects.size() == 1);
+
+        for (int i = 0; i < total_objects; i++) {
+            client.DeleteObject(
+              test_defaults::bucket_name, test_defaults::obj_prefix + std::to_string(i) + ".txt");
+        }
+    }
+
+    SECTION("Simple put test", "[gcp-connection]")
+    {
+
+        // Upload a file that does not exist locally - should fail.
+        REQUIRE(conn.put_object(non_exi_object_key, non_exi_file_name) == ENOENT);
+
+        // Check number of files with the given prefix currently in bucket
+        auto objects_iterator =
+          client.ListObjects(test_defaults::bucket_name, gcs::Prefix(test_defaults::obj_prefix));
+        int original_number_of_files =
+          std::distance(objects_iterator.begin(), objects_iterator.end());
+        std::cerr << "\n\n\n" << typeid(objects_iterator).name() << std::endl;
+        // Upload a test file
+        REQUIRE(conn.put_object(object_key, file_path) == 0);
+
+        objects_iterator =
+          client.ListObjects(test_defaults::bucket_name, gcs::Prefix(test_defaults::obj_prefix));
+
+        REQUIRE(std::distance(objects_iterator.begin(), objects_iterator.end()) ==
```

Comment:
I don't think there will be any files with this prefix from the beginning so i don't think we need the original_number_of_files variable wat do you think?